doc: add section on ironside reset handling (#24836)
nrfconnect/sdk-nrf: doc/nrf/app_dev/device_guides/nrf54h/ug_nrf54h20_ironside.rst

  * A non-zero value indicates that an error condition occurred, preventing the CPU from starting.
    Detailed information about the issue can be found in the boot report.

.. _ug_nrf54h20_ironside_se_local_domain_reset:

Local Domain Reset Handling
****************************

When a local domain experiences a reset |ISE| detects this event and triggers a global system reset to maintain system integrity.

|ISE| uses the RESETHUB peripheral to monitor reset events across all local domains.

> When a local domain reset is detected, |ISE| responds by triggering a cold boot reset (``SYS_REBOOT_COLD``) to restart the entire system.

Reviewer: This SYS_REBOOT_COLD doesn't map to any reset reason so I think it doesn't give a lot of info. Can't we instead say this triggers a global system reset and it will be marked as `SECSREQ` in RESETREAS.GLOBAL?

> Local Domain Reset Handling
> ****************************
> 
> When a local domain experiences a reset |ISE| detects this event and triggers a global system reset to maintain system integrity.
> 
> |ISE| uses the RESETHUB peripheral to monitor reset events across all local domains.
> When a local domain reset is detected, |ISE| responds by triggering a cold boot reset (``SYS_REBOOT_COLD``) to restart the entire system.
> This ensures system consistency and prevents the system from entering an inconsistent state.

Reviewer 2: We can drop some redundant sentences here IMO, for example:

```suggestion
Local Domain Reset Handling
****************************

When a local domain resets, |ISE| detects the event in the RESETHUB peripheral and triggers a global system reset.
This prevents the system from entering an inconsistent state.
```

Also, should we define what we mean by "inconsistent state"?

Reviewer 2: Should we mention that some local resets cause the device to restart in secondary mode (if configured)?

Reviewer 2: How about something like "Certain local domain resets can trigger a boot into the secondary boot mode. See (link to secondary mode section)"

  * A non-zero value indicates that an error condition occurred, preventing the CPU from starting.
    Detailed information about the issue can be found in the boot report.

.. _ug_nrf54h20_ironside_se_local_domain_reset:

Local Domain Reset Handling
****************************

When a local domain resets, |ISE| detects the event in the RESETHUB peripheral and triggers a global system reset, reported as ``SECSREQ`` in the local domain ``RESETINFO.RESETREAS.GLOBAL``.

> This prevents the system from entering an inconsistent state.

Reviewer 2: I still think this sentence is not that self-explanatory, and could maybe be rephrased/elaborated a little bit on/maybe just deleted
